head_ref/include/head_ref/HeadReference.h

```diff
@@ -83,18 +86,22 @@ class HeadReference
          */
         void publishReferences(head_ref_msgs::HeadReferenceGoal& goal);
 
-        HeadReferenceActionServer* as_;
-        std::vector < HeadReferenceActionServer::GoalHandle > goal_handles_;
+        std::unique_ptr<HeadReferenceActionServer> as_;
+        std::vector<HeadReferenceActionServer::GoalHandle> goal_handles_;
 
         bool float_topics_;
         ros::Publisher head_pub_, pan_pub_, tilt_pub_, marker_pub_;
         ros::Subscriber measurement_sub_;
 
-        tf::TransformListener* tf_listener_;
+        tf2_ros::Buffer tf_buffer_;
+        std::unique_ptr<tf2_ros::TransformListener> tf_listener_;
 
         double current_pan_, current_tilt_, goal_error_tolerance_;
         
         std::string tf_prefix_;
+        std::string frame_mount_ = "head_mount";
+        std::string frame_neck_ = "neck_tilt";
+        std::string frame_head_ = "top_kinect/openni_camera";
```

Comment:
These things look like they belong in a config file

Reply:
Check discussion above. Entire node is hardcoded. Therefore better hardcoded everything.